src/ir_def/construct.ml

```diff
@@ -162,8 +162,9 @@ let asyncE s typ_bind e typ1 =
                       eff = T.(if s = Fut then Await else Triv) }
   }
 
-let awaitE s e =
+let awaitE e =
```

Comment:
I would actually pass in the `await_sort` rather than infer it, so adjust the old code to take Type.await_sort.